src/Xrefcheck/Verify.hs

```diff
@@ -426,94 +454,103 @@ verifyReference
-            -- Calculates the seconds left until @Retry-After@ date.
-            -- Defaults to 0 if the date has already passed.
-            Date date | utcTimeToTimeSecond date >= now -> utcTimeToTimeSecond date -:- now
-            _ -> sec 0
-
-      let toRetry = any isFixable ves && numberOfRetries < ncMaxRetries cNetworking
-          currentRetryAfter = fromMaybe (ncDefaultRetryAfter cNetworking) $
-            extractRetryAfterInfo res <&> toSeconds
-
-      let moveProgress = alterOverallProgress numberOfRetries
-                       . alterProgressErrors res numberOfRetries
-
-      atomicModifyIORef' progressRef $ \VerifyProgress{..} ->
-        ( if isExternal $ locationType rLink
-          then VerifyProgress{ vrExternal =
-            let vrExternalAdvanced = moveProgress vrExternal
-            in if toRetry
-               then case pTaskTimestamp vrExternal of
-                      Just (TaskTimestamp ttc start)
-                        | currentRetryAfter +:+ now <= ttc +:+ start -> vrExternalAdvanced
-                      _ -> setTaskTimestamp currentRetryAfter now vrExternalAdvanced
-               else vrExternalAdvanced, .. }
-          else VerifyProgress{ vrLocal = moveProgress vrLocal, .. }
-        , ()
-        )
-      if toRetry
-      then do
-        threadDelay currentRetryAfter
-        retryVerification (numberOfRetries + 1) resIO
-      else return $ fmap (WithReferenceLoc fileWithReference ref) res
-
-    alterOverallProgress
+      case res of
+        Nothing -> do
+          let moveProgressOnSuccess =
+                incProgress
+                  . if errorsOccured rc
+                    then decProgressFixableErrors
+                    else id
+          modifyProgressRef now Nothing moveProgressOnSuccess
+          pure res
+        Just err ->
+          case extractRetryAfterInfo err of
+            Nothing -> do
+              let moveProgressOnUnfixable =
+                    incProgress
+                      . if errorsOccured rc
+                        then fixableToUnfixable
+                        else incProgressUnfixableErrors
+              modifyProgressRef now Nothing moveProgressOnUnfixable
+              pure res
+            Just (mbCurrentRetryAfter, counterModifier, fetchCounter, fetchCorrespondingMaximum) -> do
+              let toSeconds = \case
+                    Seconds s -> s
+                    -- Calculates the seconds left until @Retry-After@ date.
+                    -- Defaults to 0 if the date has already passed.
+                    Date date | utcTimeToTimeSecond date >= now -> utcTimeToTimeSecond date -:- now
+                    _ -> sec 0
+
+              let toRetry = rcTooManyRequestsRetries < ncMaxRetries cNetworking
+                              && rcTimeoutRetries < ncMaxTimeoutRetries cNetworking
+                  currentRetryAfter = fromMaybe (ncDefaultRetryAfter cNetworking) $
+                    fmap toSeconds mbCurrentRetryAfter
```

Comment:
I suppose `mbCurrentRetryAfter` can be not a `Maybe` now? Eventually, it is always initialized with `Just`.

Reply:
Not really: our `ExternalHttpTooManyRequests` error can be shipped with `Nothing`, when required header is missing or its contents can't be parsed properly. In this case the default value from the config is used.

Reply:
Ah right, looks good then :+1: